test/cri-containerd/removepodsandbox_test.go

```diff
@@ -0,0 +1,165 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"testing"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+type TestConfig struct {
+	name             string
+	containerName    string
+	requiredFeatures []string
+	runtimeHandler   string
+	sandboxImage     string
+	containerImage   string
+	cmd              []string
+}
+
+// Utility function to test removal of a sandbox with no containers and no previous call to stop the pod
+func runPodSandboxTestWithoutPodStop(t *testing.T, request *runtime.RunPodSandboxRequest) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	podId := runPodSandbox(t, client, ctx, request)
+	removePodSandbox(t, client, ctx, podId)
+
+	_, err := client.PodSandboxStatus(ctx, &runtime.PodSandboxStatusRequest{
+		PodSandboxId: podId,
+	})
+
+	status, ok := status.FromError(err)
+	if !ok || status.Code() != codes.NotFound {
+		t.Fatalf("PodSandboxStatus did not return expected errorStatus: %s", err)
+	}
+}
+
+// Utility function to start sandbox with one container and make sure that sandbox is removed in the end
+func runPodWithContainer(t *testing.T, client runtime.RuntimeServiceClient, ctx context.Context, tc *TestConfig) (string, string) {
+	request := getRunPodSandboxRequest(t, tc.runtimeHandler)
+	podId := runPodSandbox(t, client, ctx, &request)
+	defer removePodSandbox(t, client, ctx, podId)
+
+	containerId := createContainerInSandbox(t, client, ctx, podId, tc.containerName, tc.containerImage, tc.cmd, nil, nil, request.Config)
+
+	startContainer(t, client, ctx, containerId)
+
+	return podId, containerId
+}
+
+// Utility function to test removal of a sandbox with one container and no previous call to stop the pod or container
+func runContainerInSandboxTest(t *testing.T, tc *TestConfig) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	podId, containerId := runPodWithContainer(t, client, ctx, tc)
```

Comment:
The "go" way would be for these to be `podID` and `containerID`.  https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
And same everywhere else. For future reference mostly